### PR TITLE
Schema validations for parameters-metadata and properties-metadata

### PR DESCRIPTION
**Pull request by tal-sapan**
* Add support for default map keys in schema validation
* Fix enum schema validation
* Add check on the schema that sequence node cannot be empty
* Add validation for unallowed "datatype" key in parameters-metadata

Also use simpler matcher for validation issues for better error messages
in tests
### Checklist
- [X] Code compiles correctly
- [X] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [X] Formatting and linting run locally successfully
- [X] All tests passing
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: validations/semantic_metadata_test.go [hunk 1]
```diff
@@ -2,11 +2,9 @@ package validate
 
 import (
 	"fmt"
+	"github.com/SAP/cloud-mta/mta"
```
Comment (allaVolkov) [inline validations/semantic_metadata_test.go]: imports sequence

File: validations/semantic_metadata.go [hunk 2]
```diff
@@ -52,14 +92,14 @@ func checkParamsAndPropertiesMetadata(mta *mta.MTA, mtaNode *yaml.Node, source s
 		}
 
 		requiresNode := getPropContent(modulesNode[i], requiresYamlField)
-		issues = append(issues, checkRequiresParamsAndPropertiesMetadata(requiresNode, module.Requires)...)
+		issues = append(issues, checkRequiresParamsAndPropertiesMetadata(requiresNode, module.Requires, checkMetadata)...)
 
-		hooksNode := getPropContent(modulesNode[i], hooksYamlField)
+		hooksNode := getPropContent(modulesNode[i], "hooks")
```
Comment (allaVolkov) [inline validations/semantic_metadata.go]: why not to use constant?

Comment (tal-sapan) [reply validations/semantic_metadata.go]: Fixed

File: validations/semantic_metadata_test.go (hunk 1 above)
Comment (tal-sapan) [reply validations/semantic_metadata_test.go]: Done

**Pull request merged**